kpatch-build/kpatch-elf.h

```diff
@@ -0,0 +1,186 @@
+/*
+ * kpatch-elf.h
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA,
+ * 02110-1301, USA.
+ */
+
+#ifndef _KPATCH_ELF_H_
+#define _KPATCH_ELF_H_
+
+#include <gelf.h>
+#include "list.h"
+
+#define ERROR(format, ...) \
+	error(1, 0, "ERROR: %s: %d: " format, __FUNCTION__, __LINE__, ##__VA_ARGS__)
+
+#define log_debug(format, ...) log(DEBUG, format, ##__VA_ARGS__)
+#define log_normal(format, ...) log(NORMAL, format, ##__VA_ARGS__)
+
+#define log(level, format, ...) \
+({ \
+	if (loglevel <= (level)) \
+		printf(format, ##__VA_ARGS__); \
+})
+
+enum loglevel {
+	DEBUG,
+	NORMAL
+};
+
+extern enum loglevel loglevel;
```

Comment:
Maybe the log code could be moved to a small log.h file (though loglevel could still be defined in create-diff-object.c).